pkg/cache/sql/informer/indexer.go

```diff
@@ -0,0 +1,255 @@
+			panic("Quote characters (\") in indexer names are not supported")
+		}
+	}
+
+	tx, err := s.Begin()
+	if err != nil {
+		return nil, err
+	}
+	err = tx.Exec(fmt.Sprintf(createTableFmt, s.GetName()))
+	if err != nil {
+		return nil, err
+	}
+	err = tx.Exec(fmt.Sprintf(createIndexFmt, s.GetName()))
+	if err != nil {
+		return nil, err
+	}
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	i := &Indexer{
+		Store:    s,
+		indexers: indexers,
+	}
+	i.RegisterAfterUpsert(i.AfterUpsert)
+
+	i.deleteIndicesStmt = s.Prepare(fmt.Sprintf(deleteIndicesFmt, s.GetName()))
+	i.addIndexStmt = s.Prepare(fmt.Sprintf(addIndexFmt, s.GetName()))
+	i.listByIndexStmt = s.Prepare(fmt.Sprintf(listByIndexFmt, s.GetName()))
+	i.listKeysByIndexStmt = s.Prepare(fmt.Sprintf(listKeyByIndexFmt, s.GetName()))
+	i.listIndexValuesStmt = s.Prepare(fmt.Sprintf(listIndexValuesFmt, s.GetName()))
+
+	return i, nil
+}
+
+/* Core methods */
+
+// AfterUpsert updates indices of an object
+func (i *Indexer) AfterUpsert(key string, obj any, tx db.TXClient) error {
+	// delete all
+	err := tx.StmtExec(tx.Stmt(i.deleteIndicesStmt), key)
+	if err != nil {
+		return err
+	}
+
+	// re-insert all
+	i.indexersLock.RLock()
+	defer i.indexersLock.RUnlock()
+	for indexName, indexFunc := range i.indexers {
+		values, err := indexFunc(obj)
+		if err != nil {
+			return err
+		}
+
+		for _, value := range values {
+			err = tx.StmtExec(tx.Stmt(i.addIndexStmt), indexName, value, key)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// add afterDelete
+/* Satisfy cache.Indexer */
+
+// Index returns a list of items that match the given object on the index function
+func (i *Indexer) Index(indexName string, obj any) ([]any, error) {
+	i.indexersLock.RLock()
+	defer i.indexersLock.RUnlock()
+	indexFunc := i.indexers[indexName]
+	if indexFunc == nil {
+		return nil, fmt.Errorf("index with name %s does not exist", indexName)
+	}
+
+	values, err := indexFunc(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(values) == 0 {
+		return nil, nil
+	}
+
+	// typical case
+	if len(values) == 1 {
+		return i.ByIndex(indexName, values[0])
+	}
+
+	// atypical case - more than one value to lookup
+	// HACK: sql.Statement.Query does not allow to pass slices in as of go 1.19 - create an ad-hoc statement
+	query := fmt.Sprintf(fmt.Sprintf(selectQueryFmt, i.GetName(), strings.Repeat(", ?", len(values)-1)))
+	stmt := i.Prepare(query)
+	defer i.CloseStmt(stmt)
+	// HACK: Query will accept []any but not []string
+	params := []any{indexName}
+	for _, value := range values {
+		params = append(params, value)
+	}
+
+	rows, err := i.QueryForRows(context.TODO(), stmt, params...)
+	if err != nil {
+		return nil, err
+	}
+	return i.ReadObjects(rows, i.GetType(), i.GetShouldEncrypt())
+}
+
+// ByIndex returns the stored objects whose set of indexed values
+// for the named index includes the given indexed value
+func (i *Indexer) ByIndex(indexName, indexedValue string) ([]any, error) {
```

Comment:
Trying to use the ByIndex method here: https://github.com/tomleb/lasso-controller-notes/blob/f4b0ad310f76999edfadaf03c82f5308f6cdf3b6/simple-vai/main.go#L87-L101 and it fails with 

```
[] sql: expected 1 destination arguments in Scan, not 4
```

I have tried using an upstream `cache.SharedIndexInformer` and the same example works fine. Am I missing something in the example that would make it work?